packages/bundle/src/adaptiveCards/hooks/internal/useParseAdaptiveCardJSON.js

```diff
@@ -0,0 +1,59 @@
+import { hooks } from 'botframework-webchat-component';
+import { useCallback } from 'react';
+
+import useAdaptiveCardsPackage from '../useAdaptiveCardsPackage';
+
+const { useDirection } = hooks;
+
+function updateRTLInline(element, rtl, adaptiveCardsPackage) {
+  if (element instanceof adaptiveCardsPackage.Container) {
+    element.rtl = rtl;
+  }
+
+  // Tree traversal to add rtl boolean to child elements
+  if (element.getItemAt && element.getItemCount) {
+    const count = element.getItemCount();
+
+    for (let index = 0; index < count; index++) {
+      const child = element.getItemAt(index);
+
+      updateRTLInline(child, rtl, adaptiveCardsPackage);
+    }
+  }
+}
+
+export default function useParseAdaptiveCardJSON() {
+  const [adaptiveCardsPackage] = useAdaptiveCardsPackage();
+  const [direction] = useDirection();
+
+  const { AdaptiveCard } = adaptiveCardsPackage;
+
+  return useCallback(
+    (content, { ignoreErrors = false } = {}) => {
+      if (!content) {
+        return;
+      }
+
+      const card = new AdaptiveCard();
+      const errors = [];
+
+      // TODO: [P3] Move from "onParseError" to "card.parse(json, errors)"
```

Comment:
https://github.com/microsoft/BotFramework-WebChat/issues/1828 Add this link, and add action item to this code? Thoughts?

Reply:
For these, be sure to add `technical-debt` label

Reply:
Linked to AC schema 1.3 issue.

As we move to the new version, we probably need to update calls to API of AC, and it will touch this code.